### ISSUE TITLE
Open-ended readonly tuples can be reassigned after fixed part

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#31481]
**TypeScript Version:**  3.5.0-dev.20190518

**Search Terms:** readonly tuple rest open ended

**Code**

```ts
declare var v: readonly[number, number, ...number[]];

v[0] = 1;
v[1] = 1;
v[2] = 1; // no error
v[0 + 1] = 1;
```

**Expected behavior:**

Assigning to an element outside of the known range should give an error about a readonly index signature.

**Actual behavior:**

Reassigning `v[2]` gives no error.

**Playground Link:** http://www.typescriptlang.org/play/#src=declare%20var%20v%3A%20readonly%5Bnumber%2C%20number%2C%20...number%5B%5D%5D%3B%0D%0A%0D%0Av%5B0%5D%20%3D%201%3B%0D%0Av%5B1%5D%20%3D%201%3B%0D%0Av%5B2%5D%20%3D%201%3B%0D%0Av%5B0%20%2B%201%5D%20%3D%201%3B

**Related Issues:**
